src/keybind.cpp

```diff
@@ -387,49 +387,55 @@ void kf_CloneSelected(int limit)
 		return; // no-op
 	}
 
-	mutating_list_iterate(apsDroidLists[selectedPlayer], [limit, &sTemplate](DROID* d)
+	const DROID* droidToClone = nullptr;
+	for (const DROID* d : apsDroidLists[selectedPlayer])
 	{
-		if (d->selected)
+		if (!d->selected)
 		{
-			enumerateTemplates(selectedPlayer, [psDroid = d, &sTemplate](DROID_TEMPLATE* psTempl) {
-				if (psTempl->name.compare(psDroid->aName) == 0)
-				{
-					sTemplate = psTempl;
-					return false; // stop enumerating
-				}
-				return true;
-			});
-
-			if (!sTemplate)
-			{
-				debug(LOG_ERROR, "Cloning vat has been destroyed. We can't find the template for this droid: %s, id:%u, type:%d!", d->aName, d->id, d->droidType);
-				return IterationResult::BREAK_ITERATION;
-			}
-
-			// create a new droid army
-			for (int i = 0; i < limit; i++)
+			debug(LOG_INFO, "Nothing was selected?");
```

Comment:
This will spam the log entry for every non-selected droid prior to finding the first selected one. We probably want a bool that gets set instead if any droid is selected, and check and log once after the loop.

Reply:
Good catch, thanks. Will fix.